Source/WebCore/css/CSSRuleList.h

```diff
@@ -47,31 +48,31 @@ class CSSRuleList {
     CSSRuleList();
 };
 
-class StaticCSSRuleList final : public CSSRuleList {
+class StaticCSSRuleList final : public CSSRuleList, public RefCounted<StaticCSSRuleList> {
 public:
     static Ref<StaticCSSRuleList> create() { return adoptRef(*new StaticCSSRuleList); }
 
-    void ref() final { ++m_refCount; }
-    void deref() final;
+    void ref() final { RefCounted::ref(); }
+    void deref() final { RefCounted::deref(); }
 
     Vector<RefPtr<CSSRule>>& rules() { return m_rules; }
     
     CSSStyleSheet* styleSheet() const final { return nullptr; }
 
+    ~StaticCSSRuleList();
 private:    
     StaticCSSRuleList();
-    ~StaticCSSRuleList();
 
     unsigned length() const final { return m_rules.size(); }
     CSSRule* item(unsigned index) const final { return index < m_rules.size() ? m_rules[index].get() : nullptr; }
 
     Vector<RefPtr<CSSRule>> m_rules;
-    unsigned m_refCount;
 };
 
 // The rule owns the live list.
 template <class Rule>
 class LiveCSSRuleList final : public CSSRuleList {
+    WTF_MAKE_FAST_ALLOCATED;
```

Comment:
CSSRuleList has it so not needed.

Reply:
CSSRuleList used to have it, but I had to remove it to make StaticCSSRuleList RefCounted, so I moved it to all other CSSRuleList inheritors.

Reply:
Ah ok, i missed that